### PR TITLE
KD-1759 | Reduce NewRelic noise

### PR DESCRIPTION
**Pull request by colin-kudos**
Adds support to ignore errors based on service + npm audit fix which negates the other PR in this repo

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
   },
   "dependencies": {
     "graphql": "^14.2.1",
-    "lodash": "^4.17.13",
+    "lodash": "^4.17.19",
```
Comment (guga-kudos) [inline package.json:33]: good call in upgrading `lodash` :+1:

File: newrelicPlugin.js [hunk 2]
```diff
@@ -1,7 +1,10 @@
 /* eslint-disable import/no-unresolved */
 const newrelic = require('newrelic');
+const { getOr } = require('lodash/fp');
 const helper = require('./helper');
 
+const EXCLUDE_ERRORS_FROM_NEW_RELIC = new RegExp(getOr(null, process.env.NEW_RELIC_IGNORED_ERRORS));
```
Comment (guga-kudos) [inline newrelicPlugin.js:6]: what about creating a new method:

```js
isExcludedError(message) {
    return message.match(EXCLUDE_ERRORS_FROM_NEW_RELIC);
}
```
Using it on `didEncounterErrors`.

And then we can write some test cases to the new method and check what happens with `undefined`, `null` and `empty` values?

File: newrelicPlugin.js [hunk 3]
```diff
@@ -22,8 +24,15 @@ module.exports = class NewRelicPlugin {
...
+  isExcludedError(message, regex) {
+    return message.match(regex) !== null;
+  }
+
   // eslint-disable-next-line class-methods-use-this
   didEncounterErrors(rc) {
-    newrelic.noticeError(rc[0]);
+    if (!this.isExcludedError(rc[0].message, EXCLUDE_ERRORS_FROM_NEW_RELIC)) {
+      newrelic.noticeError(rc[0]);
+    }
```
Comment (alvin-kudos) [inline newrelicPlugin.js:36]: @guga-kudos , @talal7860 , what are your thoughts on logging the errors to stdout all the time?   Then there will be a record in the logs.

Comment (guga-kudos) [reply newrelicPlugin.js:44]: I think it's a good idea ... especially if we have to find something in the logs .

Do you want to include this in the PR? @alvin-kudos

**Pull request merged**